src/main/scala/workshop4/assignments/WikipediaSuggestion.scala

```diff
@@ -44,11 +45,7 @@ class WikipediaSuggestion extends Application {
   }
 
   def searchWikipedia(word: String): Observable[List[String]] = {
-    def urlifyWord(word: String): String = {
-      word.replace(" ", "%20")
-    }
-
-    val url = s"https://en.wikipedia.org/w/api.php?action=opensearch&search=${urlifyWord(word)}"
+    val url = s"https://en.wikipedia.org/w/api.php?action=opensearch&search=${URLEncoder.encode(word, "UTF-8")}"
```

Comment:
This works, but it actually uses HTML form encoding, which is similar, but not exactly the same as percent-encoding. I use [URI.toASCIIString()](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#toASCIIString--). This, of course, requires you to construct a `URI` object first:

```scala
new URI("https", "en.wikipedia.org", "/w/api.php", "action=opensearch&search=zoek een café", null).toASCIIString
```